sklearn/linear_model/sag.py

```diff
@@ -48,21 +51,28 @@ def get_auto_step_size(max_squared_sum, alpha_scaled, loss, fit_intercept):
     https://hal.inria.fr/hal-00860051/document
     """
     if loss in ('log', 'multinomial'):
-        # inverse Lipschitz constant for log loss
-        return 4.0 / (max_squared_sum + int(fit_intercept)
-                      + 4.0 * alpha_scaled)
+        L = (0.25 * max_squared_sum + int(fit_intercept) + alpha_scaled)
     elif loss == 'squared':
         # inverse Lipschitz constant for squared loss
-        return 1.0 / (max_squared_sum + int(fit_intercept) + alpha_scaled)
+        L = max_squared_sum + int(fit_intercept) + alpha_scaled
     else:
         raise ValueError("Unknown loss function for SAG solver, got %s "
                          "instead of 'log' or 'squared'" % loss)
+    if is_saga:
+        mun = min(2 * n_samples * alpha_scaled, L)
```

Comment:
Ok
For SAG, I used the recommendation of Mark Schmidt: [(slide 65)](http://www.birs.ca//workshops//2014/14w5003/files/schmidt.pdf)